src/blockchain/company/mod.rs

```diff
@@ -0,0 +1,412 @@
+            country_of_registration: value.country_of_registration,
+            city_of_registration: value.city_of_registration,
+            postal_address: value.postal_address,
+            email: value.email,
+            registration_number: value.registration_number,
+            registration_date: value.registration_date,
+            proof_of_registration_file: value.proof_of_registration_file,
+            logo_file: value.logo_file,
+            signatories: value.signatories,
+            public_key: value.public_key,
+            rsa_public_key: value.rsa_public_key,
+        }
+    }
+}
+
+#[derive(BorshSerialize, BorshDeserialize, Debug, Clone, PartialEq)]
+pub struct CompanyUpdateBlockData {
+    pub name: Option<String>,
+    pub email: Option<String>,
+    pub postal_address: Option<String>,
+    pub logo_file_upload_id: Option<String>,
+}
+
+#[derive(BorshSerialize, BorshDeserialize, Debug, Clone, PartialEq)]
+pub struct CompanySignCompanyBillBlockData {
+    pub bill_id: String,
+    pub block_id: u64,
+    pub block_hash: String,
+    pub operation: BillOpCode,
+}
+
+#[derive(BorshSerialize, BorshDeserialize, Debug, Clone, PartialEq)]
+pub struct CompanyAddSignatoryBlockData {
+    pub signatory: String,
+}
+
+#[derive(BorshSerialize, BorshDeserialize, Debug, Clone, PartialEq)]
+pub struct CompanyRemoveSignatoryBlockData {
+    pub signatory: String,
+}
+
+impl Block for CompanyBlock {
+    type OpCode = CompanyOpCode;
+
+    fn id(&self) -> u64 {
+        self.id
+    }
+
+    fn timestamp(&self) -> i64 {
+        self.timestamp
+    }
+
+    fn op_code(&self) -> &Self::OpCode {
+        &self.op_code
+    }
+
+    fn hash(&self) -> &str {
+        &self.hash
+    }
+
+    fn previous_hash(&self) -> &str {
+        &self.previous_hash
+    }
+
+    fn data(&self) -> &str {
+        &self.data
+    }
+
+    fn signature(&self) -> &str {
+        &self.signature
+    }
+
+    fn public_key(&self) -> &str {
+        &self.public_key
+    }
+}
+
+impl CompanyBlock {
+    fn new(
+        company_id: String,
+        id: u64,
+        previous_hash: String,
+        data: String,
+        op_code: CompanyOpCode,
+        _identity_keys: &BcrKeys,
+        company_keys: &CompanyKeys,
+        timestamp: i64,
+    ) -> Result<Self> {
+        // TODO: calculate aggregated public key from identity and company key
+        let hash = calculate_hash(
+            &id,
+            &previous_hash,
+            &data,
+            &timestamp,
+            &company_keys.public_key, // TODO: use aggregated key
+            &op_code,
+        )?;
+        // TODO: use aggregated signature from identity and company key
+        let signature = crypto::signature(&hash, &company_keys.private_key)?;
+
+        Ok(Self {
+            company_id,
+            id,
+            hash,
+            timestamp,
+            previous_hash,
+            signature,
+            public_key: company_keys.public_key.clone(), // TODO: use aggregated key
```

Comment:
Maybe we can use the pattern for to-do tasks:
TODO #TASK_NUMBER task_short_description